hudi-common/src/main/java/org/apache/hudi/common/table/view/HoodieTableFileSystemView.java

```diff
@@ -428,6 +428,7 @@ protected Option<HoodieInstant> getReplaceInstant(final HoodieFileGroupId fileGr
    * @return Stream of latest {@link FileSlice} in the partition path.
    */
   public Stream<FileSlice> fetchLatestFileSlicesIncludingInflight(String partitionPath) {
+    ensurePartitionLoadedCorrectly(partitionPath);
```

Comment:
do we need a read lock guard here? And this method is used for only testing purpose now, may need to upgrade it as top-level API.

Reply:
yes, i've moved to super class `AbstractTableFileSystemView` and added test.